lib/gadget/gadget.vala

```diff
@@ -100,12 +125,21 @@ namespace Frida.Gadget {
 			}
 			resume ();
 		} else {
-			var s = new Server (jit_enabled, gadget_range);
+
+			string listen_address;
+			uint16 listen_port;
+			string listen_uri;
+			if (!try_get_listen_address (out listen_address, out listen_port, out listen_uri)) {
+				log_error ("Invalid listen address");
+				return;
+			}
+
 			try {
-				yield s.start ();
-				server = s;
-				log_info ("Listening on TCP port 27042");
-			} catch (Error e) {
+					var s = new Server (listen_uri, jit_enabled, gadget_range);
+					yield s.start ();
+					server = s;
+					log_info ("Listening on " + listen_address + " TCP port " + listen_port.to_string());
```

Comment:
Oops, I forgot that this was a gadget-internal function. In that case just do: `log_info ("foo: %s bar: %d".printf (foo, bar));`